src/parser/TripleComponent.h

```diff
@@ -179,6 +180,22 @@ class TripleComponent {
     }
   }
 
+  /// Same as the above, but also consider the given local vocabulary. If the
+  /// string is neither in `vocabulary` nor in `localVocab`, it will be added to
+  /// `localVocab`. Therefore, we get a valid `Id` in any case.
+  template <typename Vocabulary>
+  [[nodiscard]] Id toValueId(const Vocabulary& vocabulary,
+                             std::shared_ptr<LocalVocab> localVocab) const {
+    std::optional<Id> id = toValueId(vocabulary);
+    if (!id) {
+      AD_CHECK(isString());
+      auto& newWord = getString();
+      id = Id::makeFromLocalVocabIndex(
+          localVocab->getIndexAndAddIfNotContained(std::move(newWord)));
+    }
+    return id.value();
```

Comment:
Wow, great catch + thanks for this valuable hint. I had to read up on "ref modifiers" for methods and now I got it. I have made the method `&&` now and ommited the `const` variant because both of our current use cases create the `TripleComponent' object solely to call this function on it.